docs/ref/csrf.txt

```diff
@@ -137,39 +137,29 @@ and read the token from the DOM with JavaScript:
 
     {% csrf_token %}
     <script>
-    // using jQuery
-    var csrftoken = jQuery("[name=csrfmiddlewaretoken]").val();
+    var csrftoken = document.querySelector('[name=csrfmiddlewaretoken]').value;
     </script>
 
 Setting the token on the AJAX request
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Finally, you'll have to actually set the header on your AJAX request, while
-protecting the CSRF token from being sent to other domains using
-`settings.crossDomain <https://api.jquery.com/jQuery.ajax/>`_ in jQuery 1.5.1
-and newer:
+Finally, you'll need to set the header on your AJAX request. Using the
+`fetch()`_ API:
 
 .. code-block:: javascript
 
-    function csrfSafeMethod(method) {
-        // these HTTP methods do not require CSRF protection
-        return (/^(GET|HEAD|OPTIONS|TRACE)$/.test(method));
-    }
-    $.ajaxSetup({
-        beforeSend: function(xhr, settings) {
-            if (!csrfSafeMethod(settings.type) && !this.crossDomain) {
-                xhr.setRequestHeader("X-CSRFToken", csrftoken);
-            }
-        }
+    var request = new Request(/* URL */);
+    var headers = new Headers();
+    headers.set('X-CSRFToken', csrftoken);
+    fetch(request, {
+        method: 'POST',
+        headers: headers,
+        mode: 'same-origin'  // Do not send CSRF token to another domain.
```

Comment:
Should we add `credentials: 'same-origin'`? or maybe we don't need to when we already set `mode` :thinking:

Reply:
Setting `mode` was sufficient to have it 💥 **before** sending the request when playing with this. I think that's sufficient. (Folks can add the belt and braces themselves.)